### PR TITLE
media-sound/denemo: bump to version 2.3.0

### PR DESCRIPTION
**Pull request by waebbl**
- bump to EAPI=7
- inherit xdg instead of xdg-utils
- rename doc USE flag to gtk-doc
- add subslot modifier for media-libs/aubio and media-sound/fluidsynth

**The bug below blocks this update for multilib machines and needs to be
merged first!**
Bug: https://bugs.gentoo.org/691274

Closes: https://bugs.gentoo.org/691920
Package-Manager: Portage-2.3.71, Repoman-2.3.17
Signed-off-by: Bernd Waibel <waebbl@gmail.com>

### CODE REVIEW

File: media-sound/denemo/denemo-2.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+src_test() {
+	# make check fails if used with parallel builds
+	emake -j1 check
+}
+
+src_install() {
+	DESTDIR="${D}" emake install
+	einstalldocs
```
Comment (juippis) [inline media-sound/denemo/denemo-2.3.0.ebuild]: Wouldn't calling `default` work here? It calls `emake DESTDIR="${D} install` and `einstalldocs`. I saw the ordering in emake, asking due to that.

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0.ebuild]: I can give it a try

Comment (waebbl) [reply media-sound/denemo/denemo-2.3.0.ebuild]: You were right, using `default` does work too. Changed it.

**Comment by waebbl** [comment]
@juippis Please note, the bug https://bugs.gentoo.org/691274 seems to not be solved according to b.g.o, so this ebuild will very likely fail on multilib machines, thus causing unnecessary bug reports.

**Comment by juippis** [comment]
If I get it right, it should've been solved with this commit (why did you have `Bug` instead of `Closes`?) but I can manually close it if you believe this commit fixes it.

**Comment by waebbl** [comment]
@juippis This commit doesn't fix said bug, but that bug is rather blocking this commit. See my commit message above.

**Comment by juippis** [comment]
I dont really understand. I have 
```
ABI_X86="64 32"
SYMLINK_LIB=no
LIBDIR_x86=lib
```
in my make.conf, 

```
[ebuild   R    ] media-libs/rubberband-1.8.2::gentoo  USE="-static-libs" ABI_X86="32 (64) (-x32)" 0 KiB
[ebuild  N     ] media-video/ffmpeg-4.1.4:0/56.58.58::gentoo  USE="X alsa bzip2 encode gpl iconv mp3 network opengl postproc sdl svg threads truetype vorbis x264 xcb xvid zlib (-altivec) -amr -amrenc (-appkit) -bluray -bs2b -cdio -chromaprint -chromium -codec2 -cpudetection -debug -doc -fdk -flite -fontconfig -frei0r -fribidi -gcrypt -gme -gmp -gnutls -gsm -hardcoded-tables -iec61883 -ieee1394 -jack -jpeg2k -kvazaar -ladspa -libaom -libass -libcaca -libdrm -libilbc -libressl -librtmp -libsoxr -libv4l -libxml2 -lv2 -lzma (-mipsdspr1) (-mipsdspr2) (-mipsfpu) (-mmal) -modplug -openal -opencl -openh264 -openssl -opus -oss -pic -pulseaudio -rubberband -samba -snappy -speex -srt -ssh -static-libs -test -theora -twolame -v4l -vaapi -vdpau -vpx -wavpack -webp -x265 -zeromq -zimg -zvbi" ABI_X86="32 (64) (-x32)" CPU_FLAGS_X86="aes avx avx2 fma3 mmx mmxext sse sse2 sse3 sse4_1 sse4_2 ssse3 -3dnow -3dnowext -fma4 -xop" FFTOOLS="aviocat cws2fws ffescape ffeval ffhash fourcc2pixfmt graph2dot ismindex pktdumper qt-faststart sidxindex trasher" VIDEO_CARDS="-nvidia" 10,442 KiB
[ebuild   R    ] media-sound/denemo-2.3.0::testworld  USE="alsa aubio evince fluidsynth gtk3 nls portaudio portmidi rubberband -debug -gtk-doc* -jack -static -test*" 0 KiB
```

and denemo builds and runs just fine. Do you have some sort of a fix for it, then?

**Comment by waebbl** [comment]
I don't have any of your variables in make.conf, just using abi_x86_32 selectively, using plain 17.1 profile `default/linux/amd64/17.1/systemd` (though I doubt, the systemd will make any difference, hence you should be able to use other 17.1 profiles as well) and it fails due to `/usr/lib64/pkgconfig/rubberband.pc` has a wrong `libdir` entry:

```
artus /etc # grep /lib$ /usr/lib64/pkgconfig/rubberband.pc 
libdir=${exec_prefix}/lib
```
By default, I'm also having the `jack` USE flag enabled. Just tried to install without `USE=jack` and it succeeds, but when adding `USE=jack` it fails with:

```
libtool: link: x86_64-pc-linux-gnu-gcc -O2 -pipe -march=bdver2 -fstack-protector-strong -fstack-check -pthread -I/usr/include/guile/2.0 -fdiagnostics-color=auto -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libxml2 -pthread -I/usr/include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -D_HAVE_JACK_ -D_WITH_AUBIO_4_ -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/gtksourceview-3.0 -I/usr/include/libxml2 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/evince/3.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DUSE_EVINCE -I/usr/include/alsa -D_HAVE_ALSA_ -D_HAVE_FLUIDSYNTH_ -D_HAVE_RUBBERBAND_ -D_HAVE_PORTAUDIO_ -pthread -D_HAVE_PORTMIDI_ -D_HAVE_X11_ -Wl,-O1 -Wl,--defsym=__gentoo_check_ldflags__=0 -o generate_source generate_source.o -pthread -D_HAVE_JACK_ -pthread -DUSE_EVINCE -D_HAVE_FLUIDSYNTH_ -D_HAVE_RUBBERBAND_ -D_HAVE_X11_  -Wl,--as-needed -lguile-2.0 -lgc -lxml2 -lrsvg-2 -lfontconfig -lfreetype -lsndfile -lsmf //usr/lib/libjack.so -ldb -ldl -laubio -lgtksourceview-3.0 -levview3 -levdocument3 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lgthread-2.0 -lglib-2.0 -lfluidsynth -L/usr/lib -lrubberband -lportaudio -lasound -lm -lpthread -lfftw3 -lportmidi -pthread -Wl,-rpath -Wl,//usr/lib
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: error: //usr/lib/libjack.so: incompatible target
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libxml2.so while searching for xml2
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/librsvg-2.so while searching for rsvg-2
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libfontconfig.so while searching for fontconfig
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libfreetype.so while searching for freetype
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libsndfile.so while searching for sndfile
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libdb.so while searching for db
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libdl.so while searching for dl
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libgtk-3.so while searching for gtk-3
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libgdk-3.so while searching for gdk-3
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libpangocairo-1.0.so while searching for pangocairo-1.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libpango-1.0.so while searching for pango-1.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libatk-1.0.so while searching for atk-1.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libcairo-gobject.so while searching for cairo-gobject
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libcairo.so while searching for cairo
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libgdk_pixbuf-2.0.so while searching for gdk_pixbuf-2.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libgio-2.0.so while searching for gio-2.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libgobject-2.0.so while searching for gobject-2.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libgthread-2.0.so while searching for gthread-2.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libglib-2.0.so while searching for glib-2.0
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/librubberband.so while searching for rubberband
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libportaudio.so while searching for portaudio
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libasound.so while searching for asound
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libm.so while searching for m
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libpthread.so while searching for pthread
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libfftw3.so while searching for fftw3
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libpthread.so while searching for pthread
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: warning: skipping incompatible /usr/lib/libc.so while searching for c
collect2: error: ld returned 1 exit status
```

Unfortunately, I haven't noticed the explicit link for `/usr/lib/libjack.so` in the linker command line, but only the `-L/usr/lib` and `-Wl,/usr/lib` at the very end of the command and the only file left on my system, which is needed for linking denemo is the rubberband.pc, so I thought it's because of it.

I'm going to investigate the jack failure cause then.

**Comment by waebbl** [comment]
See https://bugs.gentoo.org/692512 for this issue

**Pull request closed without merging**